docs/src/design/decoder/constraints.md

```diff
@@ -326,17 +326,23 @@ $$
 v_{dyn} = f_{dyn} \cdot ch_{dyn}  \text{ | degree} = 6
 $$
 
+When the `CALL` or `SYSCALL` operation is executed, the hash of the callee is added to the block hash table.
+
+$$
+v_{call\_or\_syscall} = (f_{call} + f_{syscall}) \cdot ch_1  \text{ | degree} = 5
+$$
+
 When `END` operation is executed, hash of the completed block is removed from the block hash table. However, we also need to differentiate between removing the first and the second child of a *join* block. We do this by looking at the next operation. Specifically, if the next operation is neither `END` nor `REPEAT` we know that another block is about to be executed, and thus, we have just finished executing the first child of a *join* block. Thus, if the next operation is neither `END` nor `REPEAT` we need to set the term for $\alpha_6$ coefficient to $1$ as shown below:
 
 $$
-u_{end} = f_{end} \cdot (bh + \alpha_6 \cdot (1 - (f_{end}' + f_{repeat}'))) \text{ | } \text{degree} = 8
+u_{end} = f_{end} \cdot (bh + \alpha_6 \cdot (1 - (f_{end}' + f_{repeat}' + f_{halt}'))) \text{ | } \text{degree} = 8
 $$
 
 Using the above definitions, we can describe the constraint for updating the block hash table as follows:
 
 > $$
 p_2' \cdot (u_{end} + 1 - f_{end}) = \\
-p_2 \cdot (v_{join} + v_{split} + v_{loop} + v_{repeat} + v_{dyn} + 1 - (f_{join} + f_{split} + f_{loop} + f_{repeat} + f_{dyn}))
+p_2 \cdot (v_{join} + v_{split} + v_{loop} + v_{repeat} + v_{dyn} + v_{call\_or\_syscall} + 1 - (f_{join} + f_{split} + f_{loop} + f_{repeat} + f_{dyn} + f_{call} + f_{syscall}))
```

Comment:
Are the $+ f_{call} + f_{syscall}$ at the end of this equation correct? Wouldn't these be redundant with `+ v_{call_or_syscall}`?

Reply:
I think so; it's the same pattern as the other ones. `v_{call_or_syscall}` is for setting `ch1` when $f_{call} + f_{syscall} = 1$, while the $+ f_{call} + f_{syscall}$ is in the `1 - ...` term to preserve the current value of `p2` when $f_{call} + f_{syscall} = 0$

Reply:
Ah yes - agreed!